cardano-cli/src/Cardano/CLI/Commands.hs

```diff
@@ -32,6 +33,8 @@ data ClientCommand
     KeyCommands KeyCmds
   | -- | Era agnostic node commands
     NodeCommands NodeCmds
+  | -- | Query commands
+    forall era. QueryCommands (QueryCmds era)
```

Comment:
`pQueryCmdsTopLevel` is instantiated by `ConwayEra`, so we can avoid the existential quantifier here if we want.

Reply:
I like this idea. I think we should define
```
type CurrentEra = ConwayEra

currentEra :: ShelleyBasedEra CurrentEra
currentEra = shelleyBasedEra
```
and use it wherever we need to reference the currently supported era.

Reply:
I think I have a way to remove all mention of era.